pkg/activator/handler/concurrency_reporter.go

```diff
@@ -41,26 +41,32 @@ func NewConcurrencyReporter(podName string, channels Channels) {
 
 	go func() {
 		outstandingRequestsPerKey := make(map[string]int32)
+
+		// Contains the number of incoming requests in the current
+		// reporting period, per key.
+		incomingRequestsPerKey := make(map[string]int32)
 		for {
 			select {
 			case event := <-channels.ReqChan:
 				switch event.EventType {
 				case ReqIn:
+					incomingRequestsPerKey[event.Key]++
```

Comment:
do we need to decrement this?

Reply:
Nope, it gets flushed after each report.